src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -108,16 +108,18 @@ seastar::future<> ClientRequest::start()
                 if (m->finish_decode()) {
                   m->clear_payload();
                 }
-                if (is_pg_op()) {
-                  return process_pg_op(pgref);
-                } else {
-                  return process_op(pgref);
-                }
+                return (is_pg_op() ?
+                        process_pg_op(pgref) :
+                        process_op(pgref));
               });
-            })
-          ).then_interruptible([pgref]() {
-            return seastar::stop_iteration::yes;
-          });
+            });
+          // keep the ordering of non-pg ops when across pg internvals
+          return (is_pg_op() ?
```

Comment:
`is_pg_op()` iterates over the op vector, IIRC. How about doing it once, at  beginning and capturing the value in the lambda?

Reply:
sure. updated and repushed.